core-api/src/main/java/com/optimizely/ab/notification/DecisionNotificationListener.java

```diff
@@ -0,0 +1,54 @@
+/****************************************************************************
+ * Copyright 2019, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+package com.optimizely.ab.notification;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.Map;
+
+public abstract class DecisionNotificationListener implements NotificationListener, DecisionNotificationListenerInterface {
+
+    /**
+     * Base notify called with var args.  This method parses the parameters and calls the abstract method.
+     *
+     * @param args - variable argument list based on the type of notification.
+     */
+    @Override
+    public final void notify(Object... args) {
+        assert (args[0] instanceof String);
```

Comment:
@aliabbasrizvi here we added multiple assert checks which will throw exception if invalid arguments are passed and it will make sure that correct arguments are passed. so I think we dont need to add this in unit tests

Reply:
We should have type specific checks i.e. for type: experiment we should check `decisionInfo` has correct experiment and variation key. Similarly for type: feature we should check `decisionInfo` has correct data.

Can you point me to where such assertions are happening? If not, then we have incomplete tests.